app/js/components/LabResultEntry.jsx

```diff
@@ -104,26 +103,100 @@ export class LabResultEntry extends PureComponent {
           </Row>
         </Grid>
       );
-    }
 
-    return (
-      <fieldset>
-        <legend>Result Details:</legend>
-        <div>
-          <EncounterFormPage
-            afterSubmitLink="/"
-            backLink="/"
-            encounterType={CONSTANTS.labResultsEncounterType}
-            formContent={observations}
-            patient={patientHeaderDetail}
-          />
-        </div>
-      </fieldset>
-    );
+      return (
+        <fieldset>
+          <legend>Result Details:</legend>
+          <Row bsClass="row result-range-header">
+            <Col sm={3} />
+            <Col sm={3} />
+            <Col sm={3} />
+            <Col sm={3}>
+              <span className="range-header-text">NORMAL RANGE</span>
+            </Col>
+          </Row>
+          <Row>
+            <EncounterFormPage
+              afterSubmitLink="/"
+              backLink="/"
+              encounterType={CONSTANTS.labResultsEncounterType}
+              formContent={observations}
+              patient={patientHeaderDetail}
+            />
+          </Row>
+        </fieldset>
+      );
+    }
+    return null;
   }
 
-  shouldRedirect() {
-    this.setState({ redirect: true });
+  renderFormContent(member) {
+    const { conceptMembers } = this.props;
+    const memberDetails = conceptMembers[member.uuid];
+    if (memberDetails) {
+      const {
+        hiNormal,
+        lowNormal,
+        lowAbsolute,
+        hiAbsolute,
+        hiCritical,
+        lowCritical,
+        units,
+      } = memberDetails;
+
+      let obsProps = {
+        concept: member.uuid,
+        path: member.uuid,
+      };
+
+      if (lowAbsolute && hiAbsolute) {
+        const minRange = minValue(lowAbsolute);
+        const maxRange = maxValue(hiAbsolute);
+        obsProps = {
+          ...obsProps,
+          validate: [minRange, maxRange],
+        };
+      }
+
+      if (hiCritical) {
+        const abnormalMaxRange = abnormalMaxValue(hiCritical);
+        obsProps = {
+          ...obsProps,
+          warn: abnormalMaxRange,
+        }
+      }
+
+      if (lowCritical) {
+        const abnormalMinRange = abnormalMinValue(lowCritical);
+        obsProps = {
+          ...obsProps,
+          warn: abnormalMinRange,
```

Comment:
We will have to handle the case where there's both a high and a low critical... I think in this case if there is both the abnormalMinRange would overwrite the previousyly defined abnormalMaxRange?